### PR TITLE
minor: consolidate `gcd` related tests

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?
N/A

## What changes are included in this PR?
The tests for  the `GCD` function exist in both `scalar.slt` and `math.slt`; it would be best to combine them.
This PR moves them all into `math.slt`.

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/sqllogictest/test_files/math.slt [hunk 1]
```diff
@@ -580,11 +580,46 @@ CREATE TABLE signed_integers(
   (4, NULL, NULL, -512, NULL, NULL)
 ;
 
-query II
-select gcd(20, 1000), lcm(20, 1000);
```
Comment (jonahgao) [inline datafusion/sqllogictest/test_files/math.slt:584]: This test has been split and moved.

File: datafusion/sqllogictest/test_files/math.slt [hunk 2]
```diff
@@ -596,20 +631,14 @@ select gcd(-9223372036854775808, 0);
 query error DataFusion error: Arrow error: Compute error: Signed integer overflow in GCD\(0, \-9223372036854775808\)
 select gcd(0, -9223372036854775808);
 
-# Test gcd with zero and negative values
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/math.slt:599]: this was combined with other examples I think

**Review by alamb — APPROVED** [review]
Thank you @jonahgao

File: datafusion/sqllogictest/test_files/math.slt (hunk 2 above)
Comment (jonahgao) [reply datafusion/sqllogictest/test_files/math.slt:599]: Yes. ` gcd(0, 0)` was duplicated with another test, and negative tests was moved too.

**Pull request merged**